packages/node_modules/@webex/webex-core/src/lib/services/service-host.js

```diff
@@ -0,0 +1,250 @@
+   * @property {string} ConstructorPTO.uri - The URI.
+   */
+
+  /**
+   * Generate a new `ServiceHost`.
+   *
+   * @param {ConstructorPTO} pto - The parameter transfer object.
+   */
+  constructor(pto) {
+    // Validate the parameter transfer object.
+    ServiceHost.validate(pto);
+
+    // Map the parameter transfer object to the class object.
+    this.catalog = pto.catalog;
+    this.default = pto.defaultUri;
+    this.hostGroup = pto.hostGroup;
+    this.id = pto.id;
+    this.priority = pto.priority;
+    this.uri = pto.uri;
+
+    // Generate flags.
+    this.failed = false;
+    this.replaced = false;
+  }
+
+  /**
+   * Determine if the `ServiceHost` is in an active state.
+   *
+   * @returns {boolean} - `true` if the service is active and usable.
+   */
+  get active() {
+    // Validate that the `ServiceHost` was marked as failed.
+    if (this.failed) {
+      return false;
+    }
+
+    // Validate that the `ServiceHost` was replaced by another `ServiceHost`.
+    if (this.replaced) {
+      return false;
+    }
+
+    // Return true once all validation has passed.
+    return true;
+  }
+
+  /**
+   * Get the cluster value.
+   *
+   * @returns {string} - The cluster value.
+   */
+  get cluster() {
+    return this.id.split(':')[2];
+  }
+
+  /**
+   * Get the group value.
+   *
+   * @returns {string} - The group value.
+   */
+  get group() {
+    return this.id.split(':')[1];
+  }
+
+  /**
+   * Get the head value.
+   *
+   * @returns {string} - The head value.
+   */
+  get head() {
+    return this.id.split(':')[0];
+  }
+
+  /**
+   * Determine if the host is local to the user's cluster.
+   *
+   * @returns {boolean} - If the host is local.
+   */
+  get local() {
+    return this.default.includes(this.hostGroup);
+  }
+
+  /**
+   * Get the service value.
+   *
+   * @returns {string} - The service value.
+   */
+  get service() {
+    return this.id.split(':')[3];
+  }
+
+  /**
+   * Get the formatted url for the host.
+   *
+   * @returns {string} - The service url.
+   */
+  get url() {
+    // Generate a url object from the default url.
+    const urlObj = Url.parse(this.default);
+
+    // Format the host of the generated url object.
+    urlObj.host = `${this.uri}${urlObj.port ? `:${urlObj.port}` : ''}`;
+
+    // Assign the formatted url to this.
+    return Url.format(urlObj);
+  }
+
+  /**
+   * The parameter transfer object for {@link setStatus}.
+   *
+   * @typedef {Object} SetStatusPTO
+   * @property {boolean} SetStatusPTO.failed - The failed status to set.
+   * @property {boolean} SetStatusPTO.replaced - The replaced status to set.
+   */
+
+  /**
+   * Set one of the status properties of the class object.
+   *
+   * @param {SetStatusPTO} pto - The parameter transfer object.
+   * @returns {this}
+   */
+  setStatus({failed, replaced}) {
+    if (failed || failed === false) {
+      this.failed = failed;
+    }
+
+    if (replaced || replaced === false) {
+      this.replaced = replaced;
```

Comment:
same as above

Reply:
This will set failed/replaced to `false` if they are not specified when calling the method. I will update the JSDoc to reflect that these are optional params.

Reply:
Used the above conditional! Thanks!